src/app/shared/services/chat/offline/chat.flow.ts

```diff
@@ -0,0 +1,207 @@
+
+    private useRouter(node: RapidProFlowExport.Node) {
+        if (node.router.operand === "@child.run.status") {
+            for (let routerCase of node.router.cases) {
+                if (routerCase.arguments && routerCase.arguments[0]) {
+                    let subscription = this.flowStatus$.subscribe((flowEvents) => {
+                        if (flowEvents.length > 0) {
+                            let latest = flowEvents[flowEvents.length - 1];
+                            if (latest.status === routerCase.arguments[0] && latest.flowId === this.childFlowId) {
+                                subscription.unsubscribe();
+                                this.childFlowId = null;
+                                this.exitUsingCategoryId(node, routerCase.category_uuid);
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    private exitUsingCategoryId(node: RapidProFlowExport.Node, matchingCategoryId: string) {
+        let matchingCategory = node.router.categories.find((cat) => cat.uuid === matchingCategoryId);
+        let matchingExit = node.exits.find((exit) => exit.uuid === matchingCategory.exit_uuid);
+        console.log("Entered node via router category ", matchingCategory);
+        this.enterNode(this.getNodeById(matchingExit.destination_uuid));
+    }
+
+    private matchHasNumberGreaterThanCase(routerCase: RapidProFlowExport.Case, incomingMsg: string): boolean {
+        let min = Number.parseFloat(routerCase.arguments[0]);
+        let inputNumber = Number.parseFloat(incomingMsg);
+        return (min !== NaN && inputNumber !== NaN && inputNumber > min);
+    }
+
+    private matchHasNumberLessThanCase(routerCase: RapidProFlowExport.Case, incomingMsg: string): boolean {
+        let max = Number.parseFloat(routerCase.arguments[0]);
+        let inputNumber = Number.parseFloat(incomingMsg);
+        return (max !== NaN && inputNumber !== NaN && inputNumber < max);
+    }
+
+    private matchHasNumberBetweenCase(routerCase: RapidProFlowExport.Case, incomingMsg: string): boolean {
+        let rangeNumbers = routerCase.arguments.map((arg) => Number.parseFloat(arg)).sort((a, b) => a - b);
+        let inputNumber = Number.parseFloat(incomingMsg);
+        if (rangeNumbers[0] !== NaN && rangeNumbers[1] !== NaN && inputNumber !== NaN) {
+            if (inputNumber > rangeNumbers[0] && inputNumber < rangeNumbers[1]) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private matchHasAnyWordCase(routerCase: RapidProFlowExport.Case, incomingMsg: string): boolean {
+        let matchStrings = [];
+        for (let arg of routerCase.arguments) {
+            matchStrings = matchStrings.concat(arg.toLowerCase().split(" "));
+        }
+        for (let matchString of matchStrings) {
+            if (incomingMsg.toLowerCase() === matchString) {
+                return true;
+            }
+        }
+    }
+
```

Comment:
Nice to have all these actions and matchers separated out, I was going to suggest whether it might be useful to include any 3rd party libs to help with the various manipulations (e.g. lodash/underscore/voca etc.), but looking at it I think the cases are probably pretty unique so maybe easiest to just handle manually.

If you want to make any of these oneliners regex is your friend here, but probably less easy on the eyes, e.g. to pass a case-insensitive flag and use a variable that would be `return new RegExp(incomingMsg,'i').test(matchstring));`

But honestly, if I really wanted one-liners I'd probably keep just do `if (incomingMsg.toLowerCase() === matchString) return true` and anger the lint fairies, so probably not the best use case for regex... but if ever useful elsewhere.

Another micro optimisation, is that you convert incomingMsg to lower case every time the loop runs, you could move it out to just convert once and save some precious zepto seconds there